Drivers/Src/CanBus.c

```diff
@@ -0,0 +1,240 @@
+/**
+ * @copyright Copyright (c) 2018-2023 UT Longhorn Racing Solar
+ * @file CANbus.c
+ * @brief
+ *
+ */
+
+#include "CanBus.h"
+
+#include "CanConfig.h"
+#include "Tasks.h"
+#include "config.h"
+#include "os.h"
+
+static OS_SEM
+    can_mail_sem4[kNumCan];  // sem4 to count how many sending hardware
+                             // mailboxes we have left (start at 3)
+static OS_SEM can_bus_receive_sem4[kNumCan];  // sem4 to count how many msgs in
+                                              // our recieving queue
+static OS_MUTEX can_bus_tx_mutex[kNumCan];    // mutex to lock tx line
+static OS_MUTEX can_bus_rx_mutex[kNumCan];    // mutex to lock Rx line
+
+/**
+ * @brief this function will be passed down to the BSP layer to trigger on RX
+ * events. Increments the receive semaphore to signal message in hardware
+ * mailbox. Do not access directly outside this driver.
+ * @param bus The CAN bus to operate on. Should be CARCAN or MOTORCAN.
+ */
+void CanBusRxHandler(Can bus) {
+    OS_ERR err = 0;
+    OSSemPost(&(can_bus_receive_sem4[bus]), OS_OPT_POST_1,
+              &err);  // increment our queue counter
+    ASSERT_OS_ERROR(err);
+}
+
+/**
+ * @brief this function will be passed down to the BSP layer to trigger on
+ * TXend. Releases hold of the mailbox semaphore (Increments it to show mailbox
+ * available). Do not access directly outside this driver.
+ * @param bus The CAN bus to operate on. Should be CARCAN or MOTORCAN.
+ */
+void CanBusTxHandler(Can bus) {
+    OS_ERR err = 0;
+    OSSemPost(&(can_mail_sem4[bus]), OS_OPT_POST_1, &err);
+    ASSERT_OS_ERROR(err);
+}
+
+// wrapper functions for the interrupt customized for each bus
+void CanBusTxHandler1() { CanBusTxHandler(kCan1); }
+
```

Comment:
Can you remove this blank line if possible/needed to be removed

Reply:
Yes